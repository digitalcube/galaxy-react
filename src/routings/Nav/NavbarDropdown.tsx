import React, { FC, PropsWithChildren } from 'react';
import { UncontrolledDropdown, DropdownToggle, DropdownMenu } from 'reactstrap';
import { Avatar, AvatarProps } from '../../components/Avatar/Avatar';

export type NavbarDropdownProps = PropsWithChildren<{
  avatar: AvatarProps;
}>;

export const NavbarDropdown: FC<NavbarDropdownProps> = ({
  children,
  avatar,
}) => {
  return (
    <UncontrolledDropdown nav inNavbar>
      <DropdownToggle
        nav
        role="button"
        data-toggle="dropdown"
        className="nav-link font-weight-bold dropdown-toggle"
      >
        <Avatar {...avatar} />
      </DropdownToggle>
      <DropdownMenu right>{children}</DropdownMenu>
    </UncontrolledDropdown>
  );
};
