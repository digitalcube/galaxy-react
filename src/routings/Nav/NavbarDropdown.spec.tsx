import React from 'react';
import { render } from '@testing-library/react';
import { Default as NavbarDropdown, InNavbar } from './NavbarDropdown.stories';

describe('NavbarDropdown', () => {
  it('to match snapshot', () => {
    const { asFragment } = render(
      <NavbarDropdown
        {...{
          ...NavbarDropdown.args,
          avatar: {
            email: 'test@example.com',
          },
        }}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('to match snapshot (in nav bar)', () => {
    const { asFragment } = render(
      <InNavbar
        {...{
          ...InNavbar.args,
          avatar: {
            email: 'test@example.com',
            rounded: false,
          },
        }}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
