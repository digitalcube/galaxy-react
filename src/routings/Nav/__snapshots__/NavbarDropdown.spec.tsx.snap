// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarDropdown to match snapshot (in nav bar) 1`] = `
<DocumentFragment>
  <nav
    class="navbar navbar-expand-lg"
  >
    <button
      aria-label="Toggle navigation"
      class="navbar-toggler"
      type="button"
    >
      <span
        class="navbar-toggler-icon"
      />
    </button>
    <div
      class="navbar-collapse flex-grow-0 collapse"
    >
      <ul
        class="ml-auto navbar-nav nav"
      >
        <li
          class="nav-item"
        >
          <a
            class="rounded font-weight-bold nav-link"
            to="#"
          >
            Guides
          </a>
        </li>
        <li
          class="dropdown nav-item"
        >
          <a
            aria-expanded="false"
            aria-haspopup="true"
            class="nav-link font-weight-bold dropdown-toggle nav-link"
            data-toggle="dropdown"
            href="#"
            role="button"
          >
            <img
              alt="Gravatar for test@example.com"
              class="react-gravatar p-0 "
              height="32"
              src="//www.gravatar.com/avatar/55502f40dc8b7c769880b10874abc9d0?d=retro&r=g&s=32"
              srcset="//www.gravatar.com/avatar/55502f40dc8b7c769880b10874abc9d0?d=retro&r=g&s=64 2x"
              width="32"
            />
          </a>
          <div
            aria-hidden="true"
            class="dropdown-menu dropdown-menu-right"
            role="menu"
            tabindex="-1"
          />
        </li>
      </ul>
    </div>
  </nav>
</DocumentFragment>
`;

exports[`NavbarDropdown to match snapshot 1`] = `
<DocumentFragment>
  <li
    class="dropdown nav-item"
  >
    <a
      aria-expanded="false"
      aria-haspopup="true"
      class="nav-link font-weight-bold dropdown-toggle nav-link"
      data-toggle="dropdown"
      href="#"
      role="button"
    >
      <img
        alt="Gravatar for test@example.com"
        class="react-gravatar p-0  rounded-circle"
        height="32"
        src="//www.gravatar.com/avatar/55502f40dc8b7c769880b10874abc9d0?d=retro&r=g&s=32"
        srcset="//www.gravatar.com/avatar/55502f40dc8b7c769880b10874abc9d0?d=retro&r=g&s=64 2x"
        width="32"
      />
    </a>
    <div
      aria-hidden="true"
      class="dropdown-menu dropdown-menu-right"
      role="menu"
      tabindex="-1"
    >
      item
    </div>
  </li>
</DocumentFragment>
`;
